src/app/store/our.reducer.ts

```diff
@@ -1,40 +1,43 @@
 import { createReducer,on } from "@ngrx/store";
-import { event_updateRotate,event_updateScale,event_updateTranslate,event_updateAdd, event_updatepropchange } from '../store/our.actions';
-import { initialState } from "./our.model";
+//import { event_updateRotate,event_updateScale,event_updateTranslate,event_updateAdd, event_updatepropchange } from '../store/our.actions';
+import { ActionTypes, OurActions } from '../store/our.actions';
+import { initialState,State } from "./our.model";
+import { UndoRedoserviceService } from '../services/undo-redoservice.service';
 
-export const counterReducer = createReducer(
-  initialState,
-  on(event_updateAdd,(state,payload)=>{
-      return{
-        ...state,
-        eventstring:payload.model.eventstring
-      }
-  }),
-  on(event_updateTranslate,(state,payload)=>{
-    return{
-      ...state,
-      eventstring:payload.model.eventstring
-    }
-}),
-on(event_updateScale,(state,payload)=>{
-  return{
-    ...state,
-    eventstring:payload.model.eventstring
-  }
-}),
-on(event_updateRotate,(state,payload)=>{
-  return{
-    ...state,
-    eventstring:payload.model.eventstring
-  }
-}),
-on(event_updatepropchange,(state,payload)=>{
-  return{
-    ...state,
-    eventstring:payload.model.eventstring
-  }
-}),
-
-);
 
+export function NewReducer(state = initialState, action: OurActions): State {
+  switch (action.type) {
+    case ActionTypes.UpdateCanvas:
+      return {
+        ...state,
+          eventstring: action.payload.eventstring,
+          undotoggle:false
+      };
+      case ActionTypes.UndoElement:
+      return {
+        ...state
+    
+      };
+      case ActionTypes.RedoElement:
+      return {
+        ...state,
+          eventstring: action.payload.eventstring,
+          undotoggle:action.payload.undotoggle
+      };
+      case ActionTypes.UndoToggle:
+        return {
+          ...state,
+            eventstring: action.payload.eventstring,
+            undotoggle:true
+        };
```

Comment:
Remove unwanted reducers. Since only `ActionTypes.UpdateCanvas` and `ActionTypes.modifiedaction` will reach the reducer, you can remove the rest of the switch cases.